packages/react/src/template.tsx

```diff
@@ -13,28 +13,27 @@ export type ComponentDefinition<T extends Record<string, JSONValue>> = {
 }
 
 /** Data state */
-export type DataState<T = never> =
+export type DataState<T = EventDataState> =
   | JSONValue
   | ReferencedComponentDataState<T>
   | ComponentDataState<T>
   | ReferencedDataState
   | ActionDataState
-  | EventDataState
   | StateDataState
   | { [key: string]: DataState<T>; $?: never }
   | DataState[]
   | T
 /** Server data state */
 export type ServerDataState = DataState<ServerEventDataState>
 
-export type ComponentDataState<T = never> = {
+export type ComponentDataState<T = EventDataState> = {
   $: 'component'
   key?: string
   component: ComponentIdentifier
   children?: DataState<T>
   props?: Record<string, DataState<T>>
 }
-type ReferencedComponentDataState<T = never> = ComponentDataState<T> & {
+type ReferencedComponentDataState<T = undefined> = ComponentDataState<T> & {
```

Comment:
fix: by turning this into `undefined`, we can skip <T> when we want the default.

Reply:
hmm, when is the referenced value supposed to be undefined? Not sure I understand this part

Reply:
LOL I thought this was working but it's obviously not. I wanted to make `ReferencedComponentDataState` completely optional but accidentally ended up passing `undefined` to ComponentDataState. Fixed